pkg/loop/internal/pb/evm/evm.proto

```diff
@@ -123,6 +123,69 @@ message Head {
   Hash parent_hash = 4;
 }
 
+message Expression {
+ oneof evaluator {
+    EXTPrimitive primitive = 1;
+    BooleanExpression boolean_expression = 2;
+  }
+}
+
+message BooleanExpression {
+  BooleanOperator boolean_operator = 1;
+  repeated Expression expression = 2;
+}
+
+message And {
+  repeated Expression expr = 1;
+}
+
+message Or {
+  repeated Expression expr = 1;
+}
+
+
+// EXTPrimitive extends general primitives with evm-specific primitives 
+message EXTPrimitive {
```

Comment:
What does `EXT` stand for? If it is just shorthand for `Extended`, then go idiom is to use `Ext`
```suggestion
// ExtPrimitive extends general primitives with evm-specific primitives 
message ExtPrimitive {
```

Reply:
I was trying to come up with a better name for this, and now I am wondering, does it actually need to exist? Or could this type and `Primitive` be combined? As in:
```diff
message Primitive {
  oneof primitive {
+    loop.Primitive general_primitive = 1;
    ContractAddress contract_address = 2;
    EventSig event_sig = 3;
    EventByWord event_by_word = 4;
    EventByTopic event_by_topic = 5;
  }
}
```